x-pack/plugins/ml/public/application/components/custom_urls/custom_url_editor/utils.ts

```diff
@@ -328,7 +401,7 @@ export async function getTestUrl(job, customUrl) {
       // attempt load the non-combined job and datafeed so they can be used in the datafeed preview
       const [{ jobs }, { datafeeds }] = await Promise.all([
         ml.getJobs({ jobId: job.job_id }),
-        ml.getDatafeeds({ datafeedId: job.datafeed_config.datafeed_id }),
+        ml.getDatafeeds({ datafeedId: job.datafeed_config?.datafeed_id ?? '' }),
```

Comment:
should we call it conditionally? 
```suggestion
        if (job.datafeed_config?.datafeed_id) {
          ml.getDatafeeds({ datafeedId: job.datafeed_config.datafeed_id  }),
        }
```

Reply:
I think this endpoint returns all datafeeds if no id is provided so it's safe and in the lines below the first datafeed from whatever response is set as `datafeedConfig` so I think it's needed.

Reply:
ah sorry, ❓ shall we pass `undefined` instead of empty string here?